main.py

```diff
@@ -36,6 +36,25 @@
 log = logging.get_logger(__name__, config=config)
 
 
+class NackException(Exception):
```

Comment:
Should we move this to a seperate file? Maybe we can even add it to the chassis. It's an easy one to add imo :)